Fix bro_int_t deprecation added in Zeek 5.1

`bro_int_t` and `bro_uint_t` are both marked as deprecated in the current Zeek master branch, replaced by `zeek_int_t` and `zeek_uint_t`. These deprecations need to be handled in the spicy-plugin to not return warnings during builds:

```
[770/937] Building CXX object src/builtin-plugins/spi...akeFiles/plugin-Zeek-Spicy.dir/src/file-analyzer.cc.o
In file included from /Users/tim/Desktop/projects/zeek/auxil/spicy-plugin/src/file-analyzer.cc:8:
/Users/tim/Desktop/projects/zeek/auxil/spicy-plugin/include/zeek-spicy/runtime-support.h:712:5: warning: 'bro_int_t' is deprecated: Remove in v6.1. Use zeek_int_t. [-Wdeprecated-declarations]
    bro_int_t bt = (it >= 0 ? it : std::numeric_limits<::bro_int_t>::max());
    ^
/Users/tim/Desktop/projects/zeek/src/zeek/util.h:82:19: note: 'bro_int_t' has been explicitly marked deprecated here
using bro_int_t [[deprecated("Remove in v6.1. Use zeek_int_t.")]] = zeek_int_t;
                  ^
In file included from /Users/tim/Desktop/projects/zeek/auxil/spicy-plugin/src/file-analyzer.cc:8:
/Users/tim/Desktop/projects/zeek/auxil/spicy-plugin/include/zeek-spicy/runtime-support.h:712:58: warning: 'bro_int_t' is deprecated: Remove in v6.1. Use zeek_int_t. [-Wdeprecated-declarations]
    bro_int_t bt = (it >= 0 ? it : std::numeric_limits<::bro_int_t>::max());
                                                         ^
/Users/tim/Desktop/projects/zeek/src/zeek/util.h:82:19: note: 'bro_int_t' has been explicitly marked deprecated here
using bro_int_t [[deprecated("Remove in v6.1. Use zeek_int_t.")]] = zeek_int_t;
                  ^
2 warnings generated.
[773/937] Building CXX object src/builtin-plugins/spi...eFiles/plugin-Zeek-Spicy.dir/src/packet-analyzer.cc.o
In file included from /Users/tim/Desktop/projects/zeek/auxil/spicy-plugin/src/packet-analyzer.cc:6:
/Users/tim/Desktop/projects/zeek/auxil/spicy-plugin/include/zeek-spicy/runtime-support.h:712:5: warning: 'bro_int_t' is deprecated: Remove in v6.1. Use zeek_int_t. [-Wdeprecated-declarations]
    bro_int_t bt = (it >= 0 ? it : std::numeric_limits<::bro_int_t>::max());
    ^
/Users/tim/Desktop/projects/zeek/src/zeek/util.h:82:19: note: 'bro_int_t' has been explicitly marked deprecated here
using bro_int_t [[deprecated("Remove in v6.1. Use zeek_int_t.")]] = zeek_int_t;
                  ^
In file included from /Users/tim/Desktop/projects/zeek/auxil/spicy-plugin/src/packet-analyzer.cc:6:
/Users/tim/Desktop/projects/zeek/auxil/spicy-plugin/include/zeek-spicy/runtime-support.h:712:58: warning: 'bro_int_t' is deprecated: Remove in v6.1. Use zeek_int_t. [-Wdeprecated-declarations]
    bro_int_t bt = (it >= 0 ? it : std::numeric_limits<::bro_int_t>::max());
                                                         ^
/Users/tim/Desktop/projects/zeek/src/zeek/util.h:82:19: note: 'bro_int_t' has been explicitly marked deprecated here
using bro_int_t [[deprecated("Remove in v6.1. Use zeek_int_t.")]] = zeek_int_t;
                  ^
2 warnings generated.
[774/937] Building CXX object src/builtin-plugins/spi...iles/plugin-Zeek-Spicy.dir/src/protocol-analyzer.cc.o
In file included from /Users/tim/Desktop/projects/zeek/auxil/spicy-plugin/src/protocol-analyzer.cc:6:
/Users/tim/Desktop/projects/zeek/auxil/spicy-plugin/include/zeek-spicy/runtime-support.h:712:5: warning: 'bro_int_t' is deprecated: Remove in v6.1. Use zeek_int_t. [-Wdeprecated-declarations]
    bro_int_t bt = (it >= 0 ? it : std::numeric_limits<::bro_int_t>::max());
    ^
/Users/tim/Desktop/projects/zeek/src/zeek/util.h:82:19: note: 'bro_int_t' has been explicitly marked deprecated here
using bro_int_t [[deprecated("Remove in v6.1. Use zeek_int_t.")]] = zeek_int_t;
                  ^
In file included from /Users/tim/Desktop/projects/zeek/auxil/spicy-plugin/src/protocol-analyzer.cc:6:
/Users/tim/Desktop/projects/zeek/auxil/spicy-plugin/include/zeek-spicy/runtime-support.h:712:58: warning: 'bro_int_t' is deprecated: Remove in v6.1. Use zeek_int_t. [-Wdeprecated-declarations]
    bro_int_t bt = (it >= 0 ? it : std::numeric_limits<::bro_int_t>::max());
                                                         ^
/Users/tim/Desktop/projects/zeek/src/zeek/util.h:82:19: note: 'bro_int_t' has been explicitly marked deprecated here
using bro_int_t [[deprecated("Remove in v6.1. Use zeek_int_t.")]] = zeek_int_t;
                  ^
2 warnings generated.
[775/937] Building CXX object src/builtin-plugins/spi...eFiles/plugin-Zeek-Spicy.dir/src/runtime-support.cc.o
In file included from /Users/tim/Desktop/projects/zeek/auxil/spicy-plugin/src/runtime-support.cc:12:
/Users/tim/Desktop/projects/zeek/auxil/spicy-plugin/include/zeek-spicy/runtime-support.h:712:5: warning: 'bro_int_t' is deprecated: Remove in v6.1. Use zeek_int_t. [-Wdeprecated-declarations]
    bro_int_t bt = (it >= 0 ? it : std::numeric_limits<::bro_int_t>::max());
    ^
/Users/tim/Desktop/projects/zeek/src/zeek/util.h:82:19: note: 'bro_int_t' has been explicitly marked deprecated here
using bro_int_t [[deprecated("Remove in v6.1. Use zeek_int_t.")]] = zeek_int_t;
                  ^
In file included from /Users/tim/Desktop/projects/zeek/auxil/spicy-plugin/src/runtime-support.cc:12:
/Users/tim/Desktop/projects/zeek/auxil/spicy-plugin/include/zeek-spicy/runtime-support.h:712:58: warning: 'bro_int_t' is deprecated: Remove in v6.1. Use zeek_int_t. [-Wdeprecated-declarations]
    bro_int_t bt = (it >= 0 ? it : std::numeric_limits<::bro_int_t>::max());
                                                         ^
/Users/tim/Desktop/projects/zeek/src/zeek/util.h:82:19: note: 'bro_int_t' has been explicitly marked deprecated here
using bro_int_t [[deprecated("Remove in v6.1. Use zeek_int_t.")]] = zeek_int_t;
                  ^
2 warnings generated.
[777/937] Building CXX object src/builtin-plugins/spi...ugin/CMakeFiles/plugin-Zeek-Spicy.dir/src/plugin.cc.o
/Users/tim/Desktop/projects/zeek/auxil/spicy-plugin/src/plugin.cc:245:42: warning: 'bro_int_t' is deprecated: Remove in v6.1. Use zeek_int_t. [-Wdeprecated-declarations]
            lval = std::numeric_limits<::bro_int_t>::max();
                                         ^
/Users/tim/Desktop/projects/zeek/src/zeek/util.h:82:19: note: 'bro_int_t' has been explicitly marked deprecated here
using bro_int_t [[deprecated("Remove in v6.1. Use zeek_int_t.")]] = zeek_int_t;
                  ^
1 warning generated.
```

--- a/include/runtime-support.h
+++ b/include/runtime-support.h
@@ -712,7 +712,7 @@ inline ::zeek::ValPtr to_val(const T& t, ::zeek::TypePtr target, const std::stri
         // can't allow this ...
         throw InvalidValue("enum values with value max_int not supported by Zeek integration", location);
 
-    bro_int_t bt = (it >= 0 ? it : std::numeric_limits<::bro_int_t>::max());
+    zeek_int_t bt = (it >= 0 ? it : std::numeric_limits<::zeek_int_t>::max());
 
     return target->AsEnumType()->GetEnumVal(bt);
 }

--- a/include/zeek-compat.h
+++ b/include/zeek-compat.h
@@ -78,6 +78,10 @@
 
 //// Import types and globals into the new namespaces.
 
+#if ZEEK_VERSION_NUMBER < 50100 // Zeek < 5.1
+using zeek_int_t = bro_int_t;
+#endif
+
 #if ZEEK_VERSION_NUMBER < 40100 // Zeek < 4.1
 namespace zeek::packet_analysis::TCP {
 using TCPSessionAdapter = ::zeek::analyzer::tcp::TCP_Analyzer;

--- a/src/plugin.cc
+++ b/src/plugin.cc
@@ -242,7 +242,7 @@ void plugin::Zeek_Spicy::Plugin::registerEnumType(
 
         if ( lval == -1 )
             // Zeek's enum can't be negative, so swap int max_int for our Undef.
-            lval = std::numeric_limits<::bro_int_t>::max();
+            lval = std::numeric_limits<::zeek_int_t>::max();
 
         etype->AddName(ns, name.c_str(), lval, true);
     }

